ios-theater/Controller/HomePageViewController.swift

```diff
@@ -53,39 +64,57 @@ class HomePageViewController: UIViewController {
       nextSection += 1
     }
     let nextIndexPath = IndexPath(item: nextItem, section: nextSection)
-    collectionView.scrollToItem(at: nextIndexPath, at: .left, animated: true)
+    bannerCollectionView.scrollToItem(at: nextIndexPath, at: .left, animated: true)
   }
 }
 
 extension HomePageViewController: UICollectionViewDataSource {
   func numberOfSections(in collectionView: UICollectionView) -> Int {
-    return largeNumberForSections
+    if collectionView == self.bannerCollectionView {
```

Comment:
You can consider separating `dataSource` and `delegate` from `ViewController`, then you won't need to this annoying logic, and will make ViewController clearer.
[FYI](https://programmingwithswift.com/how-to-separate-uitableview-datasource-and-delegate/)